packages/store-ui/src/molecules/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,131 @@
+        OpenModal
+      </Button>
+      <Modal isOpen={isOpen} testId={modalTestId} onDismiss={onDismiss}>
+        <Input testId="first-input" />
+        <Button testId="first-button" />
+      </Modal>
+    </>
+  )
+}
+
+describe('Modal', () => {
+  it('Test data-store-modal', () => {
+    const { getByTestId } = render(
+      <Modal aria-label="test modal" testId="store-modal" isOpen>
+        Foo
+      </Modal>
+    )
+
+    expect(getByTestId('store-modal')).toHaveAttribute('data-store-modal')
+  })
+
+  it('Test isOpen', () => {
+    // Check that modal won't be rendered
+    const { getByTestId } = render(<TestModal />)
+
+    expect(document.querySelector(`[data-testid="${modalTestId}"]`)).toBeNull()
+
+    const modalTriggerButton = getByTestId('trigger')
+
+    fireEvent.click(modalTriggerButton)
+
+    expect(getByTestId('store-modal')).toBeInTheDocument()
+  })
+
+  // WAI-ARIA tests
+  // https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
+  it('Test WAI-ARIA Specifications: Focus first element', () => {
+    const { getByTestId } = render(<TestModal />)
+
+    // Open the modal
+    fireEvent.click(getByTestId('trigger'))
+
+    // Check if the first tabbable is focused
+    expect(getByTestId('first-input')).toHaveFocus()
+  })
+
+  it('Test WAI-ARIA Specifications: Loop focus', () => {
+    const { getByTestId } = render(<TestModal />)
+
+    // Open the modal
+    fireEvent.click(getByTestId('trigger'))
+
+    expect(getByTestId('first-input')).toHaveFocus()
+
+    // Simulate loop back: from first to last element
+    fireEvent.keyDown(document.activeElement!, {
+      key: 'Tab',
+      shiftKey: true,
+    })
+
+    fireEvent.focus(getByTestId('sentinelStart'))
+    expect(getByTestId('first-button')).toHaveFocus()
+
+    // Simulate loop back: from last to first element
+    fireEvent.keyDown(document.activeElement!, {
+      key: 'Tab',
+    })
+
+    fireEvent.focus(getByTestId('sentinelEnd'))
+    expect(getByTestId('first-input')).toHaveFocus()
+  })
+
+  it('Test WAI-ARIA Specifications: Focus last element before the modal was opened', () => {
+    const { getByTestId } = render(<TestModal />)
+    const triggerModalButton = getByTestId('trigger')
+
+    // Focus the trigger button that's outside the modal
+    triggerModalButton.focus()
+    expect(triggerModalButton).toHaveFocus()
+
+    fireEvent.click(triggerModalButton)
+
+    // Modal focused something inside, so make sure that's not focused
+    expect(triggerModalButton).not.toHaveFocus()
+
+    // Close the modal
+    fireEvent.click(getByTestId('store-overlay'))
+
+    // Make sure that modal focused back the trigger button after close.
+    expect(triggerModalButton).toHaveFocus()
+  })
+
+  it('Test WAI-ARIA Specifications: Close when press escape', () => {
+    const { getByTestId } = render(<TestModal />)
+
+    fireEvent.click(getByTestId('trigger'))
+
+    // Press any key different from 'Escape' won't close the modal
```

Comment:
```suggestion
    // Pressing any key other than 'Escape' won't close the modal
```

Reply:
Here I also think would be good to divide into 2 tests, one for pressing escape and another for pressing a different key